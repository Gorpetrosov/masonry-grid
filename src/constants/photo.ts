
export const Photo_SubDirectory = {
    CURATED: 'curated',
    PHOTOS: 'photos',
    SEARCH: 'search'
}