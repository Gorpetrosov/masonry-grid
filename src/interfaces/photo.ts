
export interface PexelsApiResponse {
    page: number;
    per_page: number;
    photos: Photo[];
    next_page: string;
    total_results: number;
}

export interface Photo {
    id: number;
    width: number;
    height: number;
    url: string;
    photographer: string;
    photographer_url: string;
    photographer_id: number;
    avg_color: string;
    src: PhotoSrc,
    liked: false,
    alt: "Brown Rocks During Golden Hour"
}

export interface PhotoSrc {
    original: string;
    large2x: string;
    large: string;
    medium: string;
    small: string;
    portrait: string;
    landscape: string;
    tiny: string;
}