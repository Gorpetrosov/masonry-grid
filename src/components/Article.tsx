import {Link} from "react-router-dom";
import {Photo} from "../interfaces/photo.ts";

interface ArticleProps {
    photo: Photo;
}

const Article = ( {photo}: ArticleProps) => {
    return (
        <article className="cart-content">
            <div className="cart-image">
                <img src={photo.src.original} alt={photo.alt} />
            </div>
            <div className="cart-detail">
                <h3 className="text-base font-semi">Photographer {photo.photographer}</h3>
                <Link className="custom-link" target="_blank" to={photo.photographer_url}>
                    Link to {photo.photographer}'s page
                </Link>
                <p>Original Width: {photo.width} px</p>
                <p>Original Height: {photo.height} px</p>
            </div>
        </article>
    )
}

export default Article;