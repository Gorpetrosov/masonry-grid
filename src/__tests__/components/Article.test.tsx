import { render, screen } from '@testing-library/react';
import { describe, test, expect } from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import Article from "../../components/Article.tsx";
import {Photo} from "../../interfaces/photo.ts";

describe('Article Component', () => {
    const mockPhoto: Photo = {
        id: 123,
        alt: 'Brown Rocks During Golden Hour',
        photographer: 'John Doe',
        src: {
            tiny: 'test.jpg',
            original: 'test-original.jpg',
            small: '',
            medium: '',
            large: '',
            portrait: '',
            landscape: '',
            large2x: ''
        },
        photographer_url: 'https://example.com',
        width: 1920,
        height: 1080,
        url: '',
        liked: false,
        photographer_id: 1,
        avg_color: 'jjj'
    };

    test('renders all photo information correctly', () => {
        render(
            <MemoryRouter>
                <Article photo={mockPhoto} />
            </MemoryRouter>
        );

        const image = screen.getByRole('img', { name: mockPhoto.alt });
        expect(image).toHaveAttribute('src', mockPhoto.src.original);

        const heading = screen.getByRole('heading', { level: 3 });
        expect(heading).toHaveTextContent(`Photographer ${mockPhoto.photographer}`);

        expect(screen.getByText(`Original Width: ${mockPhoto.width} px`)).toBeInTheDocument();
        expect(screen.getByText(`Original Height: ${mockPhoto.height} px`)).toBeInTheDocument();
    });

    test('contains correct photographer link', () => {
        render(
            <MemoryRouter>
                <Article photo={mockPhoto} />
            </MemoryRouter>
        );

        const link = screen.getByRole('link', {
            name: `Link to ${mockPhoto.photographer}'s page`
        });

        expect(link).toHaveAttribute('href', mockPhoto.photographer_url);
        expect(link).toHaveAttribute('target', '_blank');
    });

    test('matches snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <Article photo={mockPhoto} />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });

    test('renders proper article structure', () => {
        const { container } = render(
            <MemoryRouter>
                <Article photo={mockPhoto} />
            </MemoryRouter>
        );

        const article = container.querySelector('article.cart-content');
        expect(article).toBeInTheDocument();
        expect(article?.querySelector('.cart-image')).toBeInTheDocument();
        expect(article?.querySelector('.cart-detail')).toBeInTheDocument();
    });
});