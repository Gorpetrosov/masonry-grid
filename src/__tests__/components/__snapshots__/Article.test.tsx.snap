// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article Component > matches snapshot 1`] = `
<div>
  <article
    class="cart-content"
  >
    <div
      class="cart-image"
    >
      <img
        alt="Brown Rocks During Golden Hour"
        src="test-original.jpg"
      />
    </div>
    <div
      class="cart-detail"
    >
      <h3
        class="text-base font-semi"
      >
        Photographer 
        John Doe
      </h3>
      <a
        class="custom-link"
        href="https://example.com"
        target="_blank"
      >
        Link to 
        John Doe
        's page
      </a>
      <p>
        Original Width: 
        1920
         px
      </p>
      <p>
        Original Height: 
        1080
         px
      </p>
    </div>
  </article>
</div>
`;
